tools/cloud-build/test_spot_vm.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+# Copyright 2025 "Google LLC"
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+BUILD_ID_SHORT="${BUILD_ID:0:6}"
+PROVISIONING_MODEL="SPOT"
+TERMINATION_ACTION="DELETE"
+OPTIONS_GCS_PATH=""
+FULL_INSTANCE_PREFIX="${INSTANCE_PREFIX}-${BUILD_ID_SHORT}-"
+
```

Comment:
This script depends on lot of env variables. Can you add a null check for all variables and fail early